src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/DataProviderPass.php

```diff
@@ -52,6 +53,11 @@ private function registerDataProviders(ContainerBuilder $container, string $type
         $queue = new \SplPriorityQueue();
 
         foreach ($services as $serviceId => $tags) {
+            $definition = $container->getDefinition($serviceId);
+            if (is_subclass_of($definition->getClass(), SerializerAwareDataProviderInterface::class)) {
+                $definition->addMethodCall('setSerializerLocator', [$container->getDefinition('api_platform.serializer_locator')]);
```

Comment:
You can't, but setter injection should be limited to optional dependencies or you can't assume the object is fully initialized after the call to the constructor. Setter injection should generally be avoided IMO.